modules/control/vlayer.lua

```diff
@@ -580,21 +606,32 @@ local function handle_energy_interfaces()
     -- Cap the stored energy to the allowed capacity
     if not config.unlimited_capacity and vlayer_data.storage.energy > vlayer_data.properties.capacity * mega then
         vlayer_data.storage.energy = vlayer_data.properties.capacity * mega
+
+    -- burn the trash to produce power
+    elseif vlayer_data.storage.power_items then
+        for k, v in pairs(vlayer_data.storage.power_items) do
+            local max_burning = (vlayer_data.properties.capacity * mega / 2) - vlayer_data.storage.energy
+
+            if v.count > 0 then
```

Comment:
```suggestion
            if v.count > 0 and max_burning > 0 then
```
This change is needed otherwise when the stored energy is above 50% it will start consuming power to create wood.